src/app.tsx

```diff
@@ -1,27 +1,128 @@
+
+enum AuthenticationState {
+  Authenticated,
+  Unauthenticated,
+  Checking,
+  Cancelled,
+  Error
+}
+
+/**
+ * BACKEND_HOST is configured in the root .env file. For more information,
+ * refer to the README.md.
+ */
+const AUTHENTICATION_CHECK_URL = `${BACKEND_HOST}/api/authentication/status`;
+
+const checkAuthenticationStatus = async (auth: Authentication): Promise<AuthenticationState> => {
+  /**
+   * When implementing this in your app, you might want more advanced checks.
+   * Note: You must register the provided endpoint via the Developer Portal.
+   */
+  try {
+    const token = await auth.getCanvaUserToken();
+    const res = await fetch(AUTHENTICATION_CHECK_URL, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+      method: "POST",
     });
-  };
-
-  return (
-    <div className={styles.scrollContainer}>
-      <Rows spacing="2u">
-        <Text>
-          To make changes to this app, edit the <code>src/app.tsx</code> file,
-          then close and reopen the app in the editor to preview the changes.
-        </Text>
-        <Button variant="primary" onClick={onClick} stretch>
-          Do something cool
-        </Button>
-      </Rows>
-    </div>
-  );
+    const body = await res.json();
+
+    if (body?.isAuthenticated) {
+      return AuthenticationState.Authenticated;
+    } else {
+      return AuthenticationState.Unauthenticated;
+    }
+  } catch (error) {
+    console.error(error);
+    return AuthenticationState.Error;
+  }
+};
+
+const startAuthenticationFlow = async (): Promise<AuthenticationState> => {
+  try {
+    const response = await auth.requestAuthentication();
+    switch (response.status) {
+      case "COMPLETED":
+        return AuthenticationState.Authenticated;
+      case "ABORTED":
+        console.warn("Authentication aborted by user.");
+        return AuthenticationState.Cancelled;
+      case "DENIED":
+        console.warn("Authentication denied by user", response.details);
+        return AuthenticationState.Cancelled;
+    }
+  } catch (e) {
+    console.error(e);
+    return AuthenticationState.Error;
+  }
 };
+
+export function App() {
+
+  const [authState, setAuthState] = useState<AuthenticationState>(AuthenticationState.Checking);
+
+  useEffect(() => {
+    setAuthState(AuthenticationState.Checking);
+    checkAuthenticationStatus(auth)
+        .then(status => {
+          setAuthState(status);
+        });
+  }, []);
+
+  useEffect(() => {
+    if (authState === AuthenticationState.Unauthenticated) {
+      startAuthenticationFlow()
+          .then(status => {
+            setAuthState(status);
+          });
+    }
+  }, [authState]);
+
+  if (authState === AuthenticationState.Error) {
+    console.warn(
+      "Warning: authentication not enabled on this app. Please enable auth with the instructions in README"
+    );
+    // TODO: Comment this next line out for production app
+    setAuthState(AuthenticationState.Authenticated);
```

Comment:
To be done once the authentication flow has been wired up in the Canva developer portal